cks/templates/hpa.yaml

```diff
@@ -1,5 +1,5 @@
 {{- if .Values.autoscaling.enabled }}
-apiVersion: autoscaling/v2beta1
+apiVersion: autoscaling/v2
```

Comment:
Wasn't this version introduced in 1.23? I think we need to be careful here because eks technically still has 1.19 available. 

Are these charts only supported on gcp?

I also have similar questions on @mkleene pr too. How are we testing these changes across different k8s versions?

cc: @jfuen10 @pnancarrow

Reply:
@strantalis @c-r33d we aren't testing them anyplace else since as far as I'm aware that's the only place these are used. There might be other use cases (@c3foto knows, maybe) but that's all I'm aware of.

It was introduced in [1.12](https://github.com/kubernetes/kubernetes/blob/master/CHANGELOG/CHANGELOG-1.12.md)

Reply:
This is not being used in anything that I am aware of @mkleene

Reply:
@mkleene Yeah the v2beta2 makes sense but if we wanted to introduce changes in this pull request I think we need to add some conditional logic to support backwards capability. 

https://github.com/kubernetes/kubernetes/blob/master/CHANGELOG/CHANGELOG-1.23.md#horizontalpodautoscaler-v2-graduates-to-ga